src/rules.c

```diff
@@ -521,12 +522,11 @@ void SchemaPrefixes_Free(TrieMap *t) {
   TrieMap_Free(t, freePrefixNode);
 }
 
-void SchemaPrefixes_Add(const char *prefix, StrongRef ref) {
-  size_t nprefix = strlen(prefix);
-  void *p = TrieMap_Find(SchemaPrefixes_g, (char *)prefix, nprefix);
+void SchemaPrefixes_Add(const char *prefix, size_t len, StrongRef ref) {
+  void *p = TrieMap_Find(SchemaPrefixes_g, (char *)prefix, len);
   if (p == TRIEMAP_NOTFOUND) {
     SchemaPrefixNode *node = SchemaPrefixNode_Create(prefix, ref);
-    TrieMap_Add(SchemaPrefixes_g, (char *)prefix, nprefix, node, NULL);
+    TrieMap_Add(SchemaPrefixes_g, (char *)prefix, len, node, NULL);
```

Comment:
Why should `add` be modifying the prefix? This is not safe. Does the triemap own its own data, or is it referencing external, unowned data? I think it should be refactored to be const-correct, if possible.

Reply:
It actually should be fine, it doesn't change the input. Not sure why it's not const